### PR TITLE
Diane7

### PR DESCRIPTION
**Pull request by DBB05**
Added board_page_ui and board-page server alongside changes to the main ui and server scripts. 
Interactive chart added to the Board tab.

### CODE REVIEW

**Comment by DBB05** [comment]
Great, thank you so much.

Diane Barrie | Senior Information Analyst
Public Health Scotland
E-mail: ***@***.******@***.***>

[A picture containing text  Description automatically generated]
Web<http://www.publichealthscotland.scot/> | Twitter<https://twitter.com/P_H_S_Official> | Instagram<https://www.instagram.com/publichealthscotland/>
***@***.***

File: pages/board_page.R
Comment (alicebyers5) [inline pages/board_page.R:1]: Can this file be deleted? I don't think it's used now you've created the ui and server scripts for the board page.

File: pages/board_page_server.R [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+output$waiting_list_chart <- renderPlot({
+    ggplot(data = filtered_data(),
+         aes(x = `Month/Year`, y = Value, group = Indicator)) +
+    geom_line(aes(colour= Indicator), size=1.2) +
+    theme_bw() +
+    theme(legend.text = element_text(size = 12)) +
+    theme(legend.position="bottom") +
+    theme(legend.title=element_blank()) +
+    theme(axis.text=element_text(size=12),axis.title=element_text(size=12)) +
```
Comment (alicebyers5) [inline pages/board_page_server.R:25]: The `phsstyles` package is already loaded in the `setup.R` script so you could maybe try using this to apply a theme and colour palette to your chart.

File: pages/board_page_server.R [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    # scale_x_date(date_labels = "%b %y", breaks = as.Date(c("2019-01-01", "2019-04-01",
+    #                                                        "2019-07-01", "2019-10-01",
+    scale_x_date(date_labels = "%b %y", breaks = as.Date(c("2020-01-01", "2020-04-01",
+                                                           "2020-07-01", "2020-10-01",
+                                                           "2021-01-01", "2021-04-01",
+                                                           "2021-07-01", "2021-10-01",
+                                                           "2022-01-01", "2022-04-01",
+                                                           "2022-07-01", "2022-10-01",
+                                                           "2023-01-01", "2023-04-01")))+
```
Comment (alicebyers5) [inline pages/board_page_server.R:34]: Another nice to have would be to automate this bit of code so you don't have to manually add dates when you use more data.

File: setup.R [hunk 3]
```diff
@@ -38,3 +38,5 @@ bttn_remove <-  list('select2d', 'lasso2d', 'zoomIn2d', 'zoomOut2d',
 
 # Read in data for waiting_list_chart
 Join_all_long <- read_csv("data/Join_all_long.csv",show_col_types = FALSE)
+# Read in data for waiting_list_chart
+Join_all_long2 <- read_csv("data/Join_all_long2.csv",show_col_types = FALSE)
```
Comment (alicebyers5) [inline setup.R:42]: Do you still need both of these datasets? Or could you remove one? 

It's probably also worth giving the data a clearer name as when you add more data / charts / tabs, it won't be obvious that `join_all_long` refers to waiting list data. (Unless this dataset isn't just waiting list of course!)

File: pages/board_page_server.R [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    ggplot(data = filtered_data(),
+         aes(x = `Month/Year`, y = Value, group = Indicator)) +
+    geom_line(aes(colour= Indicator), size=1.2) +
+    theme_bw() +
+    theme(legend.text = element_text(size = 12)) +
+    theme(legend.position="bottom") +
+    theme(legend.title=element_blank()) +
+    theme(axis.text=element_text(size=12),axis.title=element_text(size=12)) +
+    # scale_x_date(date_labels = "%b %y", breaks = as.Date(c("2019-01-01", "2019-04-01",
+    #                                                        "2019-07-01", "2019-10-01",
```
Comment (alicebyers5) [inline pages/board_page_server.R:27]: Could you get rid of these lines now? It'll help keep your code tidy if you remove redundant commented-out sections as you go.

**Review by alicebyers5 — APPROVED** [review]
Hi Diane,

Looks great! I've added a few comments and suggestions - they're mostly things that might make your life easier in the long run when you start adding more data/charts/tabs to the app. None are essential right now though so up to you if you want to change anything before you merge.

I've approved the pull request so you can go ahead and merge when you're ready. 

Thanks,
Alice

**Comment by DBB05** [comment]
Hi Alice,

Thank you for approving the code. All good suggestions and we can pick those up.

If you had an example of how to automate adding additional dates to the scale_x_date(date_labels that would be useful.


Thanks,
Diane

Diane Barrie | Senior Information Analyst
Public Health Scotland
E-mail: ***@***.******@***.***>

[A picture containing text  Description automatically generated]
Web<http://www.publichealthscotland.scot/> | Twitter<https://twitter.com/P_H_S_Official> | Instagram<https://www.instagram.com/publichealthscotland/>
***@***.***

**Pull request merged**